src/sdk/VoiceProfileClient.ts

```diff
@@ -0,0 +1,238 @@
+} from "./Exports";
+import { SpeechConfig, SpeechConfigImpl } from "./SpeechConfig";
+
+/**
+ * Defines VoiceProfileClient class for Speaker Recognition
+ * Handles operations from user for Voice Profile operations (e.g. createProfile, deleteProfile)
+ * @class VoiceProfileClient
+ */
+export class VoiceProfileClient {
+    protected privProperties: PropertyCollection;
+    private privAdapter: SpeakerIdMessageAdapter;
+
+    /**
+     * Gets the authorization token used to communicate with the service.
+     * @member VoiceProfileClient.prototype.authorizationToken
+     * @function
+     * @public
+     * @returns {string} Authorization token.
+     */
+    public get authorizationToken(): string {
+        return this.properties.getProperty(PropertyId.SpeechServiceAuthorization_Token);
+    }
+
+    /**
+     * Gets/Sets the authorization token used to communicate with the service.
+     * @member VoiceProfileClient.prototype.authorizationToken
+     * @function
+     * @public
+     * @param {string} token - Authorization token.
+     */
+    public set authorizationToken(token: string) {
+        Contracts.throwIfNullOrWhitespace(token, "token");
+        this.properties.setProperty(PropertyId.SpeechServiceAuthorization_Token, token);
+    }
+
+    /**
+     * The collection of properties and their values defined for this VoiceProfileClient.
+     * @member VoiceProfileClient.prototype.properties
+     * @function
+     * @public
+     * @returns {PropertyCollection} The collection of properties and their values defined for this VoiceProfileClient.
+     */
+    public get properties(): PropertyCollection {
+        return this.privProperties;
+    }
+
+    /**
+     * VoiceProfileClient constructor.
+     * @constructor
+     * @param {SpeechConfig} speechConfig - An set of initial properties for this synthesizer (authentication key, region, &c)
+     */
+    public constructor(speechConfig: SpeechConfig) {
+        const speechConfigImpl: SpeechConfigImpl = speechConfig as SpeechConfigImpl;
+        Contracts.throwIfNull(speechConfigImpl, "speechConfig");
+
+        this.privProperties = speechConfigImpl.properties.clone();
+        this.implClientSetup();
+    }
+
+    /**
+     * Create a speaker recognition voice profile
+     * @member VoiceProfileClient.prototype.createProfileAsync
+     * @function
+     * @public
+     * @param {VoiceProfileType} profileType Type of Voice Profile to be created
+     *        specifies the keyword to be recognized.
+     * @param {string} lang Language string (locale) for Voice Profile
+     * @param cb - Callback invoked once Voice Profile has been created.
+     * @param err - Callback invoked in case of an error.
+     */
+    public createProfileAsync(profileType: VoiceProfileType, lang: string, cb?: (e: VoiceProfile) => void, err?: (e: string) => void): void {
+        this.privAdapter.createProfile(profileType, lang).continueWith((promiseResult: PromiseResult<IRestResponse>) => {
```

Comment:
>continueWith [](start = 58, length = 12)

Why not call promise.on here instead of getting the result and using if statements?

Reply:
Changed